internal/provisioner/model/names.go

```diff
@@ -0,0 +1,117 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ConfigMapName returns the name of the Contour ConfigMap resource.
+func (c *Contour) ConfigMapName() string {
+	return "contour-" + c.Name
+}
+
+// ContourServiceName returns the name of the Contour Service resource.
+func (c *Contour) ContourServiceName() string {
+	return "contour-" + c.Name
+}
+
+// EnvoyServiceName returns the name of the Envoy Service resource.
+func (c *Contour) EnvoyServiceName() string {
+	return "envoy-" + c.Name
+}
+
+// ContourDeploymentName returns the name of the Contour Deployment resource.
+func (c *Contour) ContourDeploymentName() string {
+	return "contour-" + c.Name
+}
+
+// EnvoyDaemonSetName returns the name of the Envoy DaemonSet resource.
+func (c *Contour) EnvoyDaemonSetName() string {
+	return "envoy-" + c.Name
+}
+
+// LeaderElectionLeaseName returns the name of the Contour leader election Lease resource.
+func (c *Contour) LeaderElectionLeaseName() string {
+	return "leader-elect-" + c.Name
+}
+
+// ContourCertsSecretName returns the name of the Contour xDS TLS certs Secret resource.
+func (c *Contour) ContourCertsSecretName() string {
+	return c.Name + "-contourcert"
+}
+
+// EnvoyCertsSecretName returns the name of the Envoy xDS TLS certs Secret resource.
+func (c *Contour) EnvoyCertsSecretName() string {
+	return c.Name + "-envoycert"
+}
```

Comment:
I have a subsequent PR that flips these names around to use the Gateway name as a suffix rather than a prefix, for consistency. Will be done as part of directly generating the certs rather than using the Job.